Create Post and New Window

YathusanK opened the pull request:
Implemented the Create Post Use Case and implemented Interactor, Integration and View Tests for this use case. 
Also implemented the NewWindow class which opens new windows for the use cases that need it (View post, create comment and create post). This class has also been implemented into the post viewing and creating comments.

SidharthSawhney commented on src/main/java/view/display_post/PostAndCommentsView.java:
```diff
@@ -3,6 +3,9 @@
 import app.CreateCommentUseCaseBuilder;
```
For the sake of consistency of putting all View and ViewModels in one folder, we should put this file in the "view" folder and delete display_post as it increases complexity.

anbu1504 reviewed (approved):
Overall, this code is well-written and follows the principles that are outlined in this course. When examining the code, I can see that you have made great use of the factory design pattern (for example, the CreatePostUseCaseFactory) so that the structure and flow of the code relies more on abstractions rather than concretions. Using the factory design pattern also allows for the code to be easier to extend if and when needed. 

Furthermore, I see that you have followed the structure of clean architecture, as you have made good use of utilizing packages to differeniate between the different layers of the Clean Architecture diagram. For example, I see that you have defined the files in concern to the view, the files in concern to the data access layer, and the files in concern with the use case itself (the interface adapters and the application business rules) into their own separate packages and have followed those packages so that the respective files are kept as such. 

I also see that you have followed the SOLID design principles, for example, the Single Responsibility Principle. When looking at the CreatePostInteractor class, we see that indeed, the sole responsibility it has is to create a post and add that respective Post to the database. This ensures that should there be a case where debugging is needed and there is a problem with how the post is created, we can save time and go directly to the interactor file and start debugging from there. 

Please ensure that you add JavaDoc at their respective places.

SidharthSawhney commented on src/main/java/entity/PostFactory.java:
```diff
@@ -0,0 +1,11 @@
+package entity;
```
The Factory design pattern has been well implemented but we should still have an interface of the factory so that we abide by the Dependency Inversion principle

YathusanK replied:
Implemented interface

SidharthSawhney commented:
Tests are well written and isolate the interactor well! Please write JavaDoc and appropriate inline comments because there are alot of files submitted for pull request and it can be overwhelming. Package display_post under view package is not required and the files should be in view. Clean architecture is well implemented with the dependencies pointing inwards where the type of the components are their interfaces defined by their concrete implementations, for example you an attribute for the CreatePostInputBoundary in CreatePostController where the Input Boundary is concretely defined by the interactor. Single responsibility principle is implemented well with each class responsible for one task for example, the CreatePostView is responsible for the View screen and responsibility of creating posts is given to the post factory instead of statically being created in classes! Overall, well done on the code!

SidharthSawhney reviewed (changes requested):
Tests are well written and isolate the interactor well! Please write JavaDoc and appropriate inline comments because there are alot of files submitted for pull request and it can be overwhelming. Package display_post under view package is not required and the files should be in view. Clean architecture is well implemented with the dependencies pointing inwards where the type of the components are their interfaces defined by their concrete implementations, for example you an attribute for the CreatePostInputBoundary in CreatePostController where the Input Boundary is concretely defined by the interactor. Single responsibility principle is implemented well with each class responsible for one task for example, the CreatePostView is responsible for the View screen and responsibility of creating posts is given to the post factory instead of statically being created in classes! Overall, well done on the code!

YathusanK commented:
> Tests are well written and isolate the interactor well! Please write JavaDoc and appropriate inline comments because there are alot of files submitted for pull request and it can be overwhelming. Package display_post under view package is not required and the files should be in view. Clean architecture is well implemented with the dependencies pointing inwards where the type of the components are their interfaces defined by their concrete implementations, for example you an attribute for the CreatePostInputBoundary in CreatePostController where the Input Boundary is concretely defined by the interactor. Single responsibility principle is implemented well with each class responsible for one task for example, the CreatePostView is responsible for the View screen and responsibility of creating posts is given to the post factory instead of statically being created in classes! Overall, well done on the code!

Added JavaDoc and implemented the interface for the PostFactory
